pkg/manifest/runtime-manifest_test.go

```diff
@@ -211,6 +211,53 @@ func TestResolvePathParam(t *testing.T) {
 	assert.Equal(t, "person.txt", val)
 }
 
+func TestMetadataAvailableForTemplating(t *testing.T) {
+	cxt := context.NewTestContext(t)
+
+	cxt.AddTestFile("testdata/metadata-substitution.yaml", config.Name)
+	m, _ := LoadManifestFrom(cxt.Context, config.Name)
+	rm := NewRuntimeManifest(cxt.Context, ActionInstall, m)
+
+	before, _ := yaml.Marshal(m.Install[0])
+	t.Logf("Before:\n %s", before)
+	for _, step := range rm.Install {
+		rm.ResolveStep(step)
+	}
+
+	s := rm.Install[0]
+	after, _ := yaml.Marshal(s)
+	t.Logf("After:\n %s", after)
+
+	pms, ok := s.Data["exec"].(map[interface{}]interface{})
+	assert.True(t, ok)
+	cmd := pms["command"].(string)
+	assert.Equal(t, "echo \"name:HELLO version:0.1.0 description:An example Porter configuration image:jeremyrickard/porter-hello:latest\"", cmd)
+}
+
+func TestDependencyMetadataAvailableForTemplating(t *testing.T) {
+	cxt := context.NewTestContext(t)
+	cxt.AddTestFile("testdata/dep-metadata-substitution.yaml", config.Name)
+	cxt.AddTestDirectory("testdata/bundles", "bundles")
+
+	m, _ := LoadManifestFrom(cxt.Context, config.Name)
+	rm := NewRuntimeManifest(cxt.Context, ActionInstall, m)
```

Comment:
After the runtime manifest is created, `rm.bundles` needs to be populated. Since this is a unit test, you can get away with setting it with just enough data needed for your test:

```go
rm.bundles := map[string]bundle.Bundle{
		"mysql": {
			Name:"Azure MySQL",
			Description: "Azure MySQL database as a service",
			Version: "v1.0.0",
		},
	}
```

Normally that field is populated by `rm.loadDependencyDefinitions()` which reads the bundle definitions from `/cnab/app/dependencies/ALIAS/bundle.json` from inside the running bundle. But that's way more than should be setup for this test.